Sources/Fuzzilli/Corpus/MarkovCorpus.swift

```diff
@@ -0,0 +1,150 @@
+import Foundation
+
+/// Corpus & Scheduler based on 
+/// Coverage-based Greybox Fuzzing as Markov Chain paper
+/// https://mboehme.github.io/paper/TSE18.pdf
+/// Simply put, the corpus keeps track of which paths have been found, and prioritizes seeds
+/// whose path has been hit less than average. Ideally, this allows the fuzzer to prioritize
+/// less explored coverage
+
+public class MarkovCorpus: ComponentBase, Corpus {
+
+    private var allIncludedPrograms: [Program] // All programs currently in the corpus
+    private var programExecutionQueue: [Program]  // Queue of programs to be executed next, all of which hit a rare edge
+
+    private var edgeMap: [UInt64:Program]
+
+    private var totalExecs: UInt64
+
+    public init() {
+        // The corpus must never be empty. Other components, such as the ProgramBuilder, rely on this
+        self.allIncludedPrograms = []
+        self.programExecutionQueue = []
+        self.totalExecs = 0
+        self.edgeMap = [:]
+        super.init(name: "Corpus")
+
+    }
+    
+    override func initialize() {
+        // The corpus must never be empty
+        if self.isEmpty {
+            for _ in 1...5 {
+                let b = fuzzer.makeBuilder()
+                let objectConstructor = b.loadBuiltin("Object")
+                b.callFunction(objectConstructor, withArgs: [])
+                add(b.finalize())
+            }
+        }
+    }
+
+    /// Adds an individual program to the corpus
+    public func add(_ program: Program) {
+        if program.size > 0 {
+            let execution = fuzzer.execute(program)
+            guard execution.outcome == .succeeded else { return }
+            if let aspects = fuzzer.evaluator.evaluate(execution) {
```

Comment:
I changed the processInteresting function here https://github.com/googleprojectzero/fuzzilli/pull/171/files#diff-e524fe9b24bf74c95c52ee3d975a12c5352c04d22436945653f33828e38ada06R495 to call the version of `add` that already includes the aspects. This function should only be called when importing a corpus from disk, from the lead fuzzer instance, etc. Did I miss a case somewhere?

Reply:
The link doesn't seem to be working well on my end. It's line 495 of fuzzer.swift

Reply:
Right ok, makes sense. I guess my question is then if we need this API at all (see other comment below). Maybe we can remove it entirely to avoid potential bugs in the future (e.g. where someone just calls `corpus.add(program)` and doesn't realize that the program must not have been evaluated before - then things would probably break without noticeable signs that something went wrong)?

Reply:
Removed!